### ISSUE TITLE
Accessibility issues

### ISSUE THREAD

**Issue by shrilakshmishastry** [SigNoz/signoz#8910]
## Bug description

*Please describe.*  
Accessibility issues in 3 pages 
- signup and loading
- Landing page
- logs page

**Common issue**
- Decorative images are accessible to users
    eg: logo, eye image in input tag

<img width="500" height="129" alt="Image" src="https://github.com/user-attachments/assets/f64d6dfd-830b-463f-845c-35171d98a91f" />

- not use of semantic tags across components, eg: close button in alert 

**Signup and loading** 
- Error description is not announced to the screen reader when there incorrect password is entered 
- Can the description "this will create admin account" be given to the user at the beginning of flow? As it’s important to know 

<img width="522" height="182" alt="Image" src="https://github.com/user-attachments/assets/6f53312e-a1cc-42d1-9df1-663134c05101" />

- Not sure whether the loading progress is announced to  screen reader users in the loading screen  after signing up

**Landing page**
- Any reason to keep away the descriptions from keyboard users?
<img width="995" height="465" alt="Image" src="https://github.com/user-attachments/assets/51f3abaf-11a3-4425-94be-b9da99aa96ea" />

<img width="1873" height="811" alt="Image" src="https://github.com/user-attachments/assets/99669d53-e109-4444-af14-1d719d809281" />

**Logs page**
- Trapped in the search box, Esc or Tab doesn’t take the user out

<img width="994" height="284" alt="Image" src="https://github.com/user-attachments/assets/cb840a91-5cc7-4663-a63e-7f95317eb705" />
- Esc is not closing the modal 

<img width="997" height="846" alt="Image" src="https://github.com/user-attachments/assets/fb3813ec-8be1-4b2a-9d05-34e09bf0ba47" />
- 

<img width="1280" height="861" alt="Image" src="https://github.com/user-attachments/assets/96408c22-c995-42c3-b644-62e58a1552a4" />

## Expected behavior
Accessibility might not be a big concern for a developer tool, but I’m sharing a few issues anyway.
- Taking care of keyboard accessibility 
- using semantic tags
- hiding decorative images 
- Ensuring features are equally accessible to both keyboard and mouse users

## How to reproduce

1. Access pages using keyboard only
2. use voice over or any assistive technology to access the page

## Version information 
* **Signoz version**: I am trying this on development branch locally
* **Browser version**:  chrome 139
* **Your OS and version**: mac os, sequoia 15.4.1
* **Your CPU Architecture**(ARM/Intel): ARM

**Comment by srikanthccv** [comment]
Feel free to send small PRs targeting each issue.
